test/remote/gateways/remote_authorize_net_test.rb

```diff
@@ -387,7 +387,136 @@ def test_successful_verify
     response = @gateway.verify(@credit_card, @options)
     assert_success response
     assert_equal 'This transaction has been approved', response.message
-    assert_success response.responses.last, 'The void should succeed'
+    assert_equal response.responses.count, 2
+  end
+
+  def test_successful_verify_with_no_address
+    @options[:billing_address] = nil
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+    assert_equal 'This transaction has been approved', response.message
+    assert_equal response.responses.count, 2
+  end
+
+  def test_verify_with_no_zip
+    @options[:verify_amount] = 0
+    @options[:billing_address] = { zip: nil, address1: 'XYZ' }
+    response = @gateway.verify(@credit_card, @options)
+    assert_failure response
+    assert_equal 'Billing address including zip code is required for a 0 amount verify', response.message
+  end
+
+  def test_successful_verify_with_jcb_card
+    other_credit_card = credit_card('3088000000000017', brand: 'jcb')
+    response = @gateway.verify(other_credit_card, @options)
+    assert_success response
+    assert_equal 'This transaction has been approved', response.message
+    assert_equal response.responses.count, 2
+  end
+
+  def test_successful_verify_with_verify_amount_and_billing_address
+    @options[:verify_amount] = 1
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+    assert_equal 'This transaction has been approved', response.message
+  end
+
+  def test_successful_verify_with_jcb_card_and_no_billing_address
+    @options[:billing_address] = nil
+    other_credit_card = credit_card('3088000000000017', brand: 'jcb')
+    response = @gateway.verify(other_credit_card, @options)
+    assert_success response
+    assert_equal 'This transaction has been approved', response.message
+    assert_equal response.responses.count, 2
+  end
+
+  def test_successful_verify_after_store_with_gsf_value
```

Comment:
gsf is a spreedly-specific term
```suggestion
  def test_successful_verify_after_store_with_custom_verify_amount
```

Reply:
noted! code updated!